tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/codeActions/OrganizeImportsTest.xtend

```diff
@@ -0,0 +1,490 @@
+			import {A01} from "A";
+			import {A01} from "A";
+			A01;
+		''', '''
+			import {A01} from "A";
+			A01;
+		''')
+	}
+
+	@Test
+	def void testRemoveDuplicate_named02() {
+		test('''
+			import {A01} from "A";
+			import {A01 as Alias} from "A";
+			A01;
+		''', '''
+			import {A01} from "A";
+			A01;
+		''')
+	}
+
+	@Test
+	def void testRemoveDuplicate_named03() {
+		test('''
+			import {A01} from "A";
+			import {A01 as Alias} from "A";
+			Alias;
+		''', '''
+			import {A01 as Alias} from "A";
+			Alias;
+		''')
+	}
+
+	@Test
+	def void testRemoveDuplicate_named_exception01() {
+		// don't remove duplicate in case it is being used
+		test('''
+			import {A01} from "A";
+			import {A01 as Alias} from "A";
+			A01;Alias;
+		''', '''
+			import {A01} from "A";
+			import {A01 as Alias} from "A";
+			A01;Alias;
+		''')
+	}
+
+	@Test
+	def void testRemoveDuplicate_named_exception02() {
+		// don't remove duplicate in case it is being used
+		test('''
+			import {A01 as Alias1} from "A";
+			import {A01 as Alias2} from "A";
+			Alias1;Alias2;
+		''', '''
+			import {A01 as Alias1} from "A";
+			import {A01 as Alias2} from "A";
+			Alias1;Alias2;
+		''')
+	}
+
+	@Test
+	def void testRemoveDuplicate_namespace01() {
+		test('''
+			import * as N from "A";
+			import * as N from "A";
+			N.A01;
+		''', '''
+			import * as N from "A";
+			N.A01;
+		''')
+	}
+
+	@Test
+	def void testRemoveDuplicate_namespace02() {
+		test('''
+			import * as N1 from "A";
+			import * as N2 from "A";
+			N1.A01;
+		''', '''
+			import * as N1 from "A";
+			N1.A01;
+		''')
+	}
+
+	@Test
+	def void testRemoveDuplicate_namespace_exception() {
+		// don't remove duplicate in case it is being used
+		test('''
+			import * as N1 from "A";
+			import * as N2 from "A";
+			N1.A01;N2.A02;
+		''', '''
+			import * as N1 from "A";
+			import * as N2 from "A";
+			N1.A01;N2.A02;
+		''')
+	}
+
+	@Test
+	def void testRemoveDuplicate_default01() {
+		test('''
+			import D from "Def";
+			import D from "Def";
+			D;
+		''', '''
+			import D from "Def";
+			D;
+		''');
+	}
+
+	@Test
+	def void testRemoveDuplicate_default02() {
+		test('''
+			import D1 from "Def";
+			import D2 from "Def";
+			D1;
+		''', '''
+			import D1 from "Def";
+			D1;
+		''');
+	}
+
+	@Test
+	def void testRemoveDuplicate_default_exception() {
+		// don't remove duplicate in case it is being used
+		test('''
+			import D1 from "Def";
+			import D2 from "Def";
+			D1;D2;
+		''', '''
+			import D1 from "Def";
+			import D2 from "Def";
+			D1;D2;
+		''');
+	}
+
+	@Test
+	def void testSorting() {
+		test('''
+			import * as NB from "B";
+			import {A03, A01} from "A";
+			import {Def03} from "Def";
+			import {A02} from "A";
+			import * as NA from "A";
+			import "C";
+			import MyDef, {Def02} from "Def";
+			import "A";
+			import {Def01} from "Def";
+			A01;A02;A03;NA.A04;NB.B01;Def01;Def02;Def03;MyDef;
+		''', '''
+			import "C";
+			import "A";
+			import * as NA from "A";
+			import {A01} from "A";
```

Comment:
I think it's a good idea. But the old organize imports did not do it and there are a number of other cosmetic changes we might think about, e.g. omitting the trailing semicolon or using `import { A01 } from` instead of `import {A01} from`. I wanted to keep the old formatting as far as possible to not introduce unnecessary changes in all files of existing code bases, for now. Could be changed very easily at any point, of course.